examples/user_input.rs

```diff
@@ -47,10 +51,67 @@ impl Default for App {
             input: String::new(),
             input_mode: InputMode::Normal,
             messages: Vec::new(),
+            cursor_in_input: 0,
         }
     }
 }
 
+impl App {
+    fn move_cursor_left(&mut self) {
+        if !self.is_cursor_leftmost() {
+            let cursor_moved_left = self.cursor_in_input - 1;
+            self.cursor_in_input = cursor_moved_left;
+        };
+    }
+
+    fn move_cursor_right(&mut self) {
+        if !self.is_cursor_rightmost() {
+            let cursor_moved_right = self.cursor_in_input + 1;
+            self.cursor_in_input = cursor_moved_right;
+        };
+    }
+
+    fn enter_char_at_cursor(&mut self, new_char: char) {
+        if self.is_cursor_leftmost() {
+            self.input.push(new_char);
+        } else {
+            self.input.insert(self.cursor_in_input, new_char);
+        }
```

Comment:
```suggestion
        self.input.insert(self.cursor_in_input, new_char);
```

Reply:
I applied your simpler version of adding a char.